temporian/core/operators/combine.py

```diff
@@ -0,0 +1,236 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Combine operator class and public API function definitions."""
+
+from temporian.core import operator_lib
+from temporian.core.compilation import compile
+from temporian.core.data.node import (
+    EventSetNode,
+    create_node_new_features_new_sampling,
+)
+from temporian.core.operators.base import Operator
+from temporian.core.typing import EventSetOrNode
+from temporian.proto import core_pb2 as pb
+from temporian.utils.rtcheck import rtcheck
+
+MAX_NUM_ARGUMENTS = 30
+FROM_INTERSECT = "intersect"
+FROM_UNION = "union"
+FROM_FIRST = "first"
+
+
+class Combine(Operator):
+    def __init__(self, index_from: str, **inputs: EventSetNode):
+        super().__init__()
+
+        # Note: Support for dictionaries of nodes is required for
+        # serialization.
+
+        if len(inputs) < 2:
+            raise ValueError("At least two arguments should be provided")
+
+        if len(inputs) >= MAX_NUM_ARGUMENTS:
+            raise ValueError(
+                f"Too many (>{MAX_NUM_ARGUMENTS}) arguments provided"
+            )
+
+        # Attributes
+        self._index_from = index_from
+        self.add_attribute("index_from", index_from)
+
+        # inputs
+        first_input = None
+        all_unix_timestamp = True
+        for key, input in inputs.items():
+            # Check that all features are in all nodes
+            if first_input is None:
+                first_input = input
+            first_input.schema.check_compatible_features(
+                input.schema, check_order=False
+            )
+            first_input.schema.check_compatible_index(input.schema)
+
+            # Output is unix if all inputs are
+            all_unix_timestamp &= input.schema.is_unix_timestamp
+            self.add_input(key, input)
+
+        assert first_input is not None  # for static checker
+        self.add_output(
+            "output",
+            create_node_new_features_new_sampling(
+                features=first_input.schema.features,
+                indexes=first_input.indexes,
+                is_unix_timestamp=all_unix_timestamp,
+                creator=self,
+            ),
+        )
+        self.check()
+
+    @property
+    def index_from(self) -> str:
+        return self._index_from
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="COMBINE",
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="index_from",
```

Comment:
What about calling it "how" (like we do for the "join" operator), and have the same possible values?

Reply:
Like 👍🏼

Reply:
Yep.
Seeing it again from some distance I agree it's better to simplify the naming (although it only refers to the index values and doesn't join events in any sense).